perfkitbenchmarker/linux_benchmarks/gpu_pcie_bandwidth_benchmark.py

```diff
@@ -0,0 +1,248 @@
+  vm.Install('cuda_toolkit_8')
+
+
+def _ParseDeviceInfo(test_output):
+  """Parses the GPU device info from the CUDA device bandwidth test output.
+
+  Args:
+    test_output: The resulting output string from the bandwidth
+      test application.
+
+  Returns:
+    A dictionary mapping the device number to its name, for every
+    device available on the system.
+  """
+  matches = regex_util.ExtractAllMatches(r'Device\s*(\d):\s*(.*$)',
+                                         test_output, re.MULTILINE)
+  devices = {str(i[0]): str(i[1]) for i in matches}
+  return devices
+
+
+def _ParseOutputFromSingleIteration(test_output):
+  """Parses the output of the CUDA device bandwidth test.
+
+  Args:
+    test_output: The resulting output string from the bandwidth
+      test application.
+
+  Returns:
+    A dictionary containing the following values as floats:
+      * the device to host bandwidth
+      * the host to device bandwidth
+      * the device to device bandwidth
+    All units are in MB/s, as these are the units guaranteed to be output
+    by the test.
+  """
+  matches = regex_util.ExtractAllMatches(r'\d+\s+(\d+\.?\d*)', test_output)
+  results = {}
+  for i, metric in enumerate(BENCHMARK_METRICS):
+    results[metric] = float(matches[i])
+  return results
+
+
+def _CalculateMetricsOverAllIterations(result_dicts, metadata={}):
+  """Calculates stats given list of result dictionaries
+    (each item in the list represends the results from a single
+    iteration).
+
+  Args:
+    result_dicts: a list of result dictionaries. Each result dictionary
+      represents a single run of the CUDA device bandwidth test,
+      parsed by _ParseOutputFromSingleIteration().
+
+    metadata: metadata dict to be added to each Sample.
+
+  Returns:
+    a list of sample.Samples containing the device to host bandwidth,
+    host to device bandwidth, and device to device bandwidth for each
+    iteration, along with the following stats for each bandwidth type:
+      * mean
+      * min
+      * max
+      * stddev
+  """
+  samples = []
+  for metric in BENCHMARK_METRICS:
+    sequence = [x[metric] for x in result_dicts]
+    # Add a Sample for each iteration, and include the iteration number
+    # in the metadata.
+    for idx, measurement in enumerate(sequence):
+      metadata_copy = metadata.copy()
+      metadata_copy['iteration'] = idx
+      samples.append(sample.Sample(
+          metric, measurement, 'MB/s', metadata_copy))
+
+    samples.append(sample.Sample(
+        metric + ', min', min(sequence), 'MB/s', metadata))
+    samples.append(sample.Sample(
+        metric + ', max', max(sequence), 'MB/s', metadata))
+    samples.append(sample.Sample(
+        metric + ', mean', numpy.mean(sequence), 'MB/s', metadata))
+    samples.append(sample.Sample(
+        metric + ', stddev', numpy.std(sequence), 'MB/s', metadata))
+  return samples
+
+
+def _SetAndConfirmGpuClocks(vm):
+  """Sets and confirms the GPU clocks with the values provided
```

Comment:
In general first line of the doc string should end in one line.
https://google.github.io/styleguide/pyguide.html?showone=Comments#Comments